### DISCUSSION TITLE
persist data [Nov 11]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#4434] [Discord Threads]
> 🙋‍♂️ **Paul**

**Comment by o1lab** [comment]
> 🙋‍♂️ **wkw**  
#### those belong to meta data which is stored in the DB (the one you specify in `NC_DB`)  
#  
> 🙋‍♂️ **wkw**  
#### if you've specified it, then even you restart your container, your data is still here. (since it is stored in NC_DB)  
#  
> 🙋‍♂️ **wkw**  
#### if it is not specified, we'll create a sqlite file inside the container. so if you haven't mounted the volume, then once you restart the container, your data will be lost (since the container got destroyed, so does the sqlite file).  
#  
> 🙋‍♂️ **Paul**  
#### Ok, thank you. I'll check NC_DB 🙂  
#  
> 🙋‍♂️ **Paul**  
#### I'm using : NC_DB="mssql://host:port?u=user&p=password&d=database" to set up the database when the container starts but it doesn't work.   
#### Are there any print/lettering that can affect this (notably in the password section) ?   
#### Are there any other connection chain that can persist data ?   
#### I'm already able connect thought the configuration panel 🙂  
#  
> 🙋‍♂️ **Paul**  
#### Also, do I need to encapsulate some of the item between ""?   
#### Thank you in advance 🙂  
#  
> 🙋‍♂️ **wkw**  
#### when u start the app, any errors saying connection to NC_DB failed?  
#  
> 🙋‍♂️ **wkw**  
#### and can u show how you deploy? using cloud shell? if so, can u show the full command? redact some sensitive data first  
#  
> 🙋‍♂️ **Paul**  
#### When I start the app, here is the message that I got :  
![image0](https://cdn.discordapp.com/attachments/1032594239844007947/1032694731265822731/unknown.png)  
#  
> 🙋‍♂️ **Paul**  
#### I deploy with the Cloud Run service on GCP.   
#### As environment variable, I set : "NC_DB="mssql://host:port?u=user&p=password&d=database""  
#  
> 🙋‍♂️ **wkw**  
#### if u r using cli, then try   
####    
#### --set-env-vars  "NC_DB=mssql://host:port?u=user&p=password&d=database"  
#  
> 🙋‍♂️ **Paul**  
#### I do not use CLI. I do it directly in GCP through the panel :   
####    
#### (I replace host, port, user, password and database with real data of course)  
![image0](https://cdn.discordapp.com/attachments/1032594239844007947/1032700384055140413/unknown.png)  
#  
> 🙋‍♂️ **Paul**  
#### Is the NC_DB works on every versions of nocodb ? maybe, I'm not using the last one 😉  
#  
> 🙋‍♂️ **wkw**  
#### Then no double quotes r needed  
#  
> 🙋‍♂️ **wkw**  
#### Yes  
#  
> 🙋‍♂️ **wkw**  
#### Which version r u using?  
#  
> 🙋‍♂️ **wkw**  
#### And is it working fine without NC_DB?  
#  
> 🙋‍♂️ **Paul**  
#### I'm using 0.90.8.    
#### Yes it is working fine without NC_DB 🙂  
#  
> 🙋‍♂️ **wkw**  
#### is it working when you input it without quotes?  
#  
> 🙋‍♂️ **wkw**  
#### maybe check the logs to hv more insights  
#  
> 🙋‍♂️ **wkw**  
#### besides, suggest to use the latest version as we've multiple bug fix releases after that  
#  
> 🙋‍♂️ **Paul**  
#### No it's not working without quotes as well 🥲  
#  
> 🙋‍♂️ **wkw**  
#### then u may try latest version / check the logs  
#  
> 🙋‍♂️ **Paul**  
#### Thank you 🙂  
#  
> 🙋‍♂️ **Paul**  
#### Hello, for info, it works pointing NC_DB to another mysql db.   
#### I think there might be an issue with the variable NC_DB and MsSQL 🙂  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1032594239844007947)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
